File: C/Algorithms/Sorting/BinarySearch.c
#include<stdio.h>
#include<stdlib.h>

int compare(const void *a, const void *b){
    return ( *(int*)a - *(int*)b );
}

// int binarySearch(int arr[], int n, int target){

//     int s = 0;
//     int e = n - 1;

//     while(s <= e){
//         int m = s + (e - s)/2;

//         if(arr[m] == target){
//             return m;
//         }

//         else if(arr[m] > target){
//             e = m - 1;
//         }

//         else{
//             s = m + 1;
//         }
//     }
//     return -1;
// }

void main(){
    int arr[] = {23,12,45,67,34,56,89,11,90};
    int n = sizeof(arr)/sizeof(arr[0]);
    int target = 11;

    qsort(arr, n, sizeof(int), compare);
    int *p;
    p = bsearch(&target, arr, n, sizeof(int), compare);
    // int ans = binarySearch(arr,n,target);
    
    if(p){
        printf("Target Element found\n");
    }
    else{
        printf("Target Element not found \n");
    }
}

File: Java/GFG Problems/Array/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        // int[] arr = {1,2,3,4,5};
        // int target = 4;
        int[] arr = {11,22,33,44,55};
        int target = 445;
        int ans = binarySearch(arr,target);
        System.out.println(ans);
    }

    static int binarySearch(int[] arr, int K) {
        int s = 0;
        int e = arr.length - 1 ;

        while(s <= e){
            int m = s + (e - s) / 2;
            if(arr[m] == K){
                return m;
            }
            if(arr[m] < K){
                s = m + 1;
            }
            else{
                e = m - 1;
            }
        }
        return -1;
    }
}
